web/src/containers/activity/ContractorResource/ContractorResourceForm.js

```diff
@@ -32,29 +32,17 @@ const ContractorResourceForm = ({
 
   const handle = {
     changeDesc: useCallback(handleChange(index, 'desc'), []),
-    changeHourlyHours: apdFFYs.reduce(
-      (acc, ffy) => ({
-        ...acc,
-        [ffy]: useCallback(handleHourlyChange(index, ffy, 'hours'), [apdFFYs])
-      }),
```

Comment:
This is a bug. The number and order of hooks needs to be deterministic - it can't change between renders, or else React loses track of what it's supposed to return. Here, the number of callbacks is based on how many federal fiscal years the APD has. If you add a year, that increases the number of hook calls, and React throws an exception for that.

Reply:
![whoa](https://media1.giphy.com/media/ToMjGpnXBTw7vnokxhu/source.gif)